theano/gpuarray/subtensor.py

```diff
@@ -1182,3 +1182,37 @@ def infer_shape(self, node, shapes):
             diag_size = T.minimum(dim1, dim2)
         out_shape.append(diag_size)
         return [tuple(out_shape)]
+
+
+class GpuAllocDiag(Subtensor):
+    __props__ = ("offset",)
+
+    def __init__(self, offset=0, axis1=0, axis2=1, view=False):
+        self.offset = offset
+
+    def make_node(self, _x):
+        ctx_name = infer_context_name(_x)
+        x = as_gpuarray_variable(_x, ctx_name)
+
+        if x.ndim != 1:
+            raise ValueError('AllocDiag argument must be a vector!', x)
+        return gof.Apply(self, [x], [x.type.__class__(dtype=x.dtype)()])
```

Comment:
Just call x.type(), otherwise, you are missing things like broadcastable and context.